R/gage_locations.R

```diff
@@ -251,18 +280,55 @@ get_hydrologic_locations <- function(all_gages, ref_locations, hydrologic_locati
   linked_gages <- select(no_location, provider_id) |>
     mutate(id = seq_len(n())) |>
     left_join(select(linked_gages_dedup, 
-                     id, COMID, REACHCODE, REACH_meas), 
+                     id, COMID, REACHCODE, REACH_meas, nhdpv2_totdasqkm, offset), 
               by = "id")
   
   all_gages$nhdpv2_REACHCODE[update_index] <- linked_gages$REACHCODE
   all_gages$nhdpv2_REACH_measure[update_index] <- linked_gages$REACH_meas
   all_gages$nhdpv2_COMID[update_index] <- linked_gages$COMID
+  all_gages$nhdpv2_totdasqkm[update_index] <- linked_gages$nhdpv2_totdasqkm
+  all_gages$nhdpv2_link_source[update_index] <- rep("https://github.com/internetofwater/ref_gages", nrow(linked_gages))
+  all_gages$nhdpv2_offset_m[update_index] <- linked_gages$offset
+
+  all_gages$nhdpv2_totdasqkm <- round(all_gages$nhdpv2_totdasqkm, digits = 1)
+  all_gages$drainage_area_sqkm <- round(all_gages$drainage_area_sqkm, digits = 1)
+    
+  all_gages$da_diff <- all_gages$nhdpv2_totdasqkm - all_gages$drainage_area_sqkm
   
-  all_gages
+  add_offset(all_gages, nhdpv2_fline)
 }
 
-add_mainstems <- function(gage_hydrologic_locations, mainstems, vaa) {
+add_offset <- function(all_gages, nhdpv2_fline) {
+  
+  missing_offset <- all_gages |>
+    filter(is.na(nhdpv2_offset_m) & !is.na(nhdpv2_REACHCODE))
+  
+  missing_offset <- sf::st_transform(missing_offset, sf::st_crs(nhdpv2_fline))
+  
+  new_indexes <- hydroloom::index_points_to_lines(nhdpv2_fline, sf::st_geometry(missing_offset),
+                                                  search_radius =  units::set_units(1000, "meters"),
+                                                  ids = as.integer(missing_offset$nhdpv2_COMID))
+  
+  missing_offset$point_id <- seq_len(nrow(missing_offset))
+  
+  missing_offset <- left_join(missing_offset, new_indexes, by = "point_id")
+  
+  missing_offset$nhdpv2_offset_m <- missing_offset$offset
+  
+  missing_offset <- sf::st_transform(missing_offset, sf::st_crs(all_gages))
+  
+  missing_offset$nhdpv2_REACHCODE[is.na(missing_offset$nhdpv2_offset_m)] <- NA
+  missing_offset$nhdpv2_REACH_measure[is.na(missing_offset$nhdpv2_offset_m)] <- NA
+  missing_offset$nhdpv2_COMID[is.na(missing_offset$nhdpv2_offset_m)] <- NA
+  missing_offset$nhdpv2_totdasqkm[is.na(missing_offset$nhdpv2_offset_m)] <- NA
+  missing_offset$nhdpv2_link_source[is.na(missing_offset$nhdpv2_offset_m)] <- NA
```

Comment:
could do this in 1 line:
```suggestion
  missing_offset[is.na(missing_offset$nhdpv2_offset_m), c('nhdpv2_REACHCODE', 'nhdpv2_REACH_measure','nhdpv2_COMID','nhdpv2_totdasqkm','nhdpv2_link_source')] <- NA
```